### ISSUE TITLE
Questions about soft matching

### ISSUE THREAD

**Issue by ICE-Bruce** [jingyi0000/R1-VL#8]
Firstly, great work on the StepGRPO framework! I'm particularly interested in the Step-wise Reasoning Accuracy Reward (StepRAR) and have a couple of questions:

1. I noticed that you utilized GPT-4 to extract the key step from the annotated answer. Will the GPT-4 also be used for model-generated reasoning steps to extract the key answer?
2. How is the soft matching between model-generated reasoning steps and pre-extracted key steps implemented? Is it based on a pre-trained language model or some rule-based matching?

**Comment by jingyi0000** [comment]
Hi, thanks for your interest!

(1) GPT-4 is only used to pre-extract key steps from the Chain-of-Thought (CoT) annotations. It is **not** used to extract key steps from the model-generated reasoning.

(2) The matching between the model-generated reasoning and the pre-extracted key steps is done through a simple presence check—that is, we determine whether each key step appears in the generated reasoning.

We will release the training code and the data used for training as soon as possible.

**Comment by ICE-Bruce** [comment]
Thank you~
